int32[2] intersection_point_2d
float64[3] intersection_point_3d